src/test/ui/anonymous-higher-ranked-lifetime.stderr

```diff
@@ -40,7 +40,7 @@ LL |     f2(|_: (), _: ()| {});
    |     expected signature of `fn(&'a (), &()) -> _`
 ...
 LL | fn f2<F>(_: F) where F: for<'a> Fn(&'a (), &()) {}
-   | ----------------------------------------------- required by `f2`
+   |    --                             ------------- required by this bound in `f2`
```

Comment:
This is an odd span that indicates to me that we have an off-by-one error somewhere earlier (parsing? lowering?)

Reply:
Fixed